### PR TITLE
seperated classes to follow srp

### PR DESCRIPTION
**Pull request by MichelleChernyi**
seperated classes encounter into encounter and interactables manager to follow srp with @devan-srinivasan

### CODE REVIEW

File: src/main/usecases/InteractablesManager.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package usecases;
```
Comment (devan-srinivasan) [inline src/main/usecases/InteractablesManager.java:1]: Looks good. Im glad we lightened up Encounter!

File: src/main/usecases/Encounter.java [hunk 2]
```diff
@@ -298,24 +204,24 @@ public boolean isCompleted() {
     public String getHelp(Player player) {
         System.out.println("All commands will be of the form -- command: [param_type]=[param_name]");
         if (doingGeneric) {
-            if (this.genericPool.get(currGenericIndex) instanceof Spinnable) {
```
Comment (devan-srinivasan) [inline src/main/usecases/Encounter.java:301]: as long as all these big changes work, we are good to go. did you test it?

Comment (MichelleChernyi) [reply src/main/usecases/Encounter.java:301]: yes of course :)

Comment (devan-srinivasan) [reply src/main/usecases/Encounter.java:301]: okay merging.

**Pull request merged**